server/src/test/java/org/apache/druid/server/http/MetadataResourceTest.java

```diff
@@ -236,6 +246,117 @@ public void testGetAllSegmentsIncludingRealtime()
     Assert.assertEquals(new SegmentStatusInCluster(realTimeSegments[1], false, null, 40L, true), resultList.get(5));
   }
 
+  @Test
+  public void testGetUnusedSegmentsInDataSource()
+  {
+    Mockito.doAnswer(mockIterateAllUnusedSegmentsForDatasource())
+        .when(segmentsMetadataManager)
+        .iterateAllUnusedSegmentsForDatasource(
+            ArgumentMatchers.any(),
+            ArgumentMatchers.any(),
+            ArgumentMatchers.any(),
+            ArgumentMatchers.any());
+
+    // test with null datasource name - fails with expected bad datasource name error
+    DruidException e = Assert.assertThrows(
+        DruidException.class,
+        () -> metadataResource.getUnusedSegmentsInDataSource(request, null, null, -1, null)
+    );
+    Assert.assertEquals(DruidException.Category.INVALID_INPUT, e.getCategory());
+    Assert.assertEquals("dataSource name must be non-empty", e.getMessage());
```

Comment:
You could replace this with the exception helper matcher:

```suggestion
    DruidExceptionMatcher.invalidInput().expectMessageIs(
        "dataSource name must be non-empty"
    ).assertThrowsAndMatches(
        () -> metadataResource.getUnusedSegmentsInDataSource(request, "", null, -1, null)
    );
```